app/src/main/java/org/mozilla/fenix/search/toolbar/ToolbarView.kt

```diff
@@ -111,8 +111,14 @@ class ToolbarView(
 
     fun update(searchState: SearchFragmentState) {
         if (!isInitialized) {
-            view.url = searchState.query
-            view.setSearchTerms(searchState.session?.searchTerms ?: "")
+            view.url = searchState.pastedText ?: searchState.query
+
+            /* Only set the search terms if pasted text is null so that the search term doesn't
+            overwrite pastedText when view enters `editMode` */
```

Comment:
Normally when we go to `editMode`, searchTerms overwrites the "url" value. This is done in `BrowserToolbar`. Because we don't want to show either search terms nor the url, we need to make sure the search terms are only set if a user didn't paste something!